### ISSUE TITLE
DNSSEC validation fails for a record without a DS record nor a SOA record

### ISSUE THREAD

**Issue by vincentbernat** [systemd/systemd#22131]
**systemd version the issue has been seen with**
 > master

**Used distribution**
 > Debian unstable

**Linux kernel version used** (`uname -a`)
 > `Linux neo 5.15.0-2-amd64 #1 SMP Debian 5.15.5-2 (2021-12-18) x86_64 GNU/Linux`

**CPU architecture issue was seen on**
 > x86-64

**Expected behaviour you didn't see**
 > Solving steering.nextdns.io when DNSSEC is enabled should succeed

**Unexpected behaviour you saw**
 > Resolution fails

**Steps to reproduce the problem**

1. Configure resolved with:
```
[Resolve]
DNS=8.8.8.8
DNSSEC=yes
```
2. Restart systemd-resolved
3. Query: `resolvectl query dns.nextdns.io`

resolved seems to reach the conclusion there is no DS record for `steering.nextdns.io` (proved by NSEC), however, it returns "bogus" as the status for SOA (which does not exist either). Selected trace:

```text
Requesting SOA to validate transaction 53398 (steering.nextdns.io, unsigned non-SOA/NS RRset <steering.nextdns.io IN A 37.252.225.79>).
Requesting DS (→ steering.nextdns.io) to validate transaction 9851 (steering.nextdns.io, unsigned empty SOA/NS response).
Proved NODATA via NSEC/NSEC3 for transaction 55150 (steering.nextdns.io IN DS)
Found verdict for lookup steering.nextdns.io IN DS: secure
Validating response from transaction 9851 (steering.nextdns.io IN SOA).
Found verdict for lookup steering.nextdns.io IN SOA: bogus
DNSSEC validation failed for question steering.nextdns.io IN SOA: no-signature
Regular transaction 9851 for <steering.nextdns.io IN SOA> on scope dns on enp9s0/* now complete with <dnssec-failed> from network (unsigned; non-confidential).
Auxiliary DNSSEC RR query failed validation: no-signature
DNSSEC validation failed for question steering.nextdns.io IN A: no-signature
Regular transaction 53398 for <steering.nextdns.io IN A> on scope dns on enp9s0/* now complete with <dnssec-failed> from network (unsigned; non-confidential).
Auxiliary DNSSEC RR query failed validation: no-signature
DNSSEC validation failed for question steering.nextdns.io IN AAAA: no-signature
Regular transaction 47901 for <steering.nextdns.io IN AAAA> on scope dns on enp9s0/* now complete with <dnssec-failed> from network (unsigned; non-confidential).
Freeing transaction 53398.
```

Here is the whole debug output from resolved:

```text
 18:07 ❱ sudo ./systemd-resolved
Positive Trust Anchors:
. IN DS 20326 8 2 e06d44b80b8f1d39a95c0b0d7c65d08458e880409bbc683457104237c7f8ec8d
Negative trust anchors: home.arpa 10.in-addr.arpa 16.172.in-addr.arpa 17.172.in-addr.arpa 18.172.in-addr.arpa 19.172.in-addr.arpa 20.172.in-addr.arpa 21.172.in-addr.arpa 22.172.in-addr.arpa 23.172.in-addr.arpa 24.172.in-addr.arpa 25.172.in-addr.arpa 26.172.in-addr.arpa 27.172.in-addr.arpa 28.172.in-addr.arpa 29.172.in-addr.arpa 30.172.in-addr.arpa 31.172.in-addr.arpa 168.192.in-addr.arpa d.f.ip6.arpa corp home internal intranet lan local private test
Using system hostname 'neo'.
Setting log level to debug.
Sent message type=method_return sender=n/a destination=:1.6291 path=n/a interface=n/a member=n/a cookie=29 reply_cookie=2 signature=n/a error-name=n/a error-message=n/a
Got message type=method_call sender=:1.6292 destination=org.freedesktop.resolve1 path=/org/freedesktop/resolve1 interface=org.freedesktop.resolve1.Manager member=ResolveHostname cookie=2 reply_cookie=0 signature=isit error-name=n/a error-message=n/a
idn2_lookup_u8: dns.nextdns.io → dns.nextdns.io
Sent message type=method_call sender=n/a destination=org.freedesktop.DBus path=/org/freedesktop/DBus interface=org.freedesktop.DBus member=GetConnectionUnixProcessID cookie=30 reply_cookie=0 signature=s error-name=n/a error-message=n/a
Got message type=method_return sender=org.freedesktop.DBus destination=:1.6290 path=n/a interface=n/a member=n/a cookie=19 reply_cookie=30 signature=u error-name=n/a error-message=n/a
D-Bus hostname resolution request from client PID 3134880 (resolvectl) with UID 500
Looking up RR for dns.nextdns.io IN A.
Looking up RR for dns.nextdns.io IN AAAA.
Sent message type=method_call sender=n/a destination=org.freedesktop.DBus path=/org/freedesktop/DBus interface=org.freedesktop.DBus member=AddMatch cookie=31 reply_cookie=0 signature=s error-name=n/a error-message=n/a
Sent message type=method_call sender=n/a destination=org.freedesktop.DBus path=/org/freedesktop/DBus interface=org.freedesktop.DBus member=GetNameOwner cookie=32 reply_cookie=0 signature=s error-name=n/a error-message=n/a
Got message type=method_return sender=org.freedesktop.DBus destination=:1.6290 path=n/a interface=n/a member=n/a cookie=21 reply_cookie=32 signature=s error-name=n/a error-message=n/a
Switching to DNS server 172.18.254.200 for interface wg0.
Switching to DNS server 2a01:cb00:59:f102::1 for interface enp9s0.
Cache miss for dns.nextdns.io IN AAAA
Firing regular transaction 22015 for <dns.nextdns.io IN AAAA> scope dns on enp9s0/* (validate=yes).
Using feature level UDP+EDNS0+DO for transaction 22015.
Using DNS server 2a01:cb00:59:f102::1 for transaction 22015.
Announcing packet size 1452 in egress EDNS(0) packet.
Emitting UDP, link MTU is 1500, socket MTU is 0, minimal MTU is 60
Sending query packet with id 22015 of size 66.
Cache miss for dns.nextdns.io IN A
Firing regular transaction 19450 for <dns.nextdns.io IN A> scope dns on enp9s0/* (validate=yes).
Using feature level UDP+EDNS0+DO for transaction 19450.
Using DNS server 2a01:cb00:59:f102::1 for transaction 19450.
Announcing packet size 1452 in egress EDNS(0) packet.
Emitting UDP, link MTU is 1500, socket MTU is 0, minimal MTU is 60
Sending query packet with id 19450 of size 66.
Got message type=method_return sender=org.freedesktop.DBus destination=:1.6290 path=n/a interface=n/a member=n/a cookie=20 reply_cookie=31 signature=n/a error-name=n/a error-message=n/a
Match type='signal',sender='org.freedesktop.DBus',path='/org/freedesktop/DBus',interface='org.freedesktop.DBus',member='NameOwnerChanged',arg0=':1.6292' successfully installed.
Received dns UDP packet of size 228, ifindex=2, ttl=0, fragsize=0, sender=2a01:cb00:59:f102::1, destination=2a01:cb00:59:f102:3905:14fb:c301:5421
Processing incoming packet of size 228 on transaction 22015 (rcode=SUCCESS).
Verified we get a response at feature level UDP+EDNS0+DO from DNS server 2a01:cb00:59:f102::1.
Requesting DNSKEY to validate transaction 22015 (dns.nextdns.io, RRSIG with key tag: 34505).
Cache miss for nextdns.io IN DNSKEY
Firing regular transaction 22752 for <nextdns.io IN DNSKEY> scope dns on enp9s0/* (validate=yes).
Using feature level UDP+EDNS0+DO for transaction 22752.
Using DNS server 2a01:cb00:59:f102::1 for transaction 22752.
Announcing packet size 1452 in egress EDNS(0) packet.
Emitting UDP, link MTU is 1500, socket MTU is 0, minimal MTU is 60
Sending query packet with id 22752 of size 62.
Received dns UDP packet of size 204, ifindex=2, ttl=0, fragsize=0, sender=2a01:cb00:59:f102::1, destination=2a01:cb00:59:f102:3905:14fb:c301:5421
Processing incoming packet of size 204 on transaction 19450 (rcode=SUCCESS).
Requesting DNSKEY to validate transaction 19450 (dns.nextdns.io, RRSIG with key tag: 34505).
Received dns UDP packet of size 305, ifindex=2, ttl=0, fragsize=0, sender=2a01:cb00:59:f102::1, destination=2a01:cb00:59:f102:3905:14fb:c301:5421
Processing incoming packet of size 305 on transaction 22752 (rcode=SUCCESS).
Requesting DS to validate transaction 22752 (nextdns.io, DNSKEY with key tag: 34505).
Cache miss for nextdns.io IN DS
Firing regular transaction 50553 for <nextdns.io IN DS> scope dns on enp9s0/* (validate=yes).
Using feature level UDP+EDNS0+DO for transaction 50553.
Using DNS server 2a01:cb00:59:f102::1 for transaction 50553.
Announcing packet size 1452 in egress EDNS(0) packet.
Emitting UDP, link MTU is 1500, socket MTU is 0, minimal MTU is 60
Sending query packet with id 50553 of size 62.
Requesting DS to validate transaction 22752 (nextdns.io, DNSKEY with key tag: 2371).
Received dns UDP packet of size 249, ifindex=2, ttl=0, fragsize=0, sender=2a01:cb00:59:f102::1, destination=2a01:cb00:59:f102:3905:14fb:c301:5421
Processing incoming packet of size 249 on transaction 50553 (rcode=SUCCESS).
Requesting DNSKEY to validate transaction 50553 (nextdns.io, RRSIG with key tag: 28626).
Cache miss for io IN DNSKEY
Firing regular transaction 28936 for <io IN DNSKEY> scope dns on enp9s0/* (validate=yes).
Using feature level UDP+EDNS0+DO for transaction 28936.
Using DNS server 2a01:cb00:59:f102::1 for transaction 28936.
Announcing packet size 1452 in egress EDNS(0) packet.
Emitting UDP, link MTU is 1500, socket MTU is 0, minimal MTU is 60
Sending query packet with id 28936 of size 54.
Received dns UDP packet of size 893, ifindex=2, ttl=0, fragsize=0, sender=2a01:cb00:59:f102::1, destination=2a01:cb00:59:f102:3905:14fb:c301:5421
Processing incoming packet of size 893 on transaction 28936 (rcode=SUCCESS).
Requesting DS to validate transaction 28936 (io, DNSKEY with key tag: 28626).
Cache miss for io IN DS
Firing regular transaction 6338 for <io IN DS> scope dns on enp9s0/* (validate=yes).
Using feature level UDP+EDNS0+DO for transaction 6338.
Using DNS server 2a01:cb00:59:f102::1 for transaction 6338.
Announcing packet size 1452 in egress EDNS(0) packet.
Emitting UDP, link MTU is 1500, socket MTU is 0, minimal MTU is 60
Sending query packet with id 6338 of size 54.
Requesting DS to validate transaction 28936 (io, DNSKEY with key tag: 57355).
Requesting DS to validate transaction 28936 (io, DNSKEY with key tag: 2790).
Received dns UDP packet of size 366, ifindex=2, ttl=0, fragsize=0, sender=2a01:cb00:59:f102::1, destination=2a01:cb00:59:f102:3905:14fb:c301:5421
Processing incoming packet of size 366 on transaction 6338 (rcode=SUCCESS).
Requesting DNSKEY to validate transaction 6338 (io, RRSIG with key tag: 9799).
Cache miss for . IN DNSKEY
Firing regular transaction 63618 for <. IN DNSKEY> scope dns on enp9s0/* (validate=yes).
Using feature level UDP+EDNS0+DO for transaction 63618.
Using DNS server 2a01:cb00:59:f102::1 for transaction 63618.
Announcing packet size 1452 in egress EDNS(0) packet.
Emitting UDP, link MTU is 1500, socket MTU is 0, minimal MTU is 60
Sending query packet with id 63618 of size 51.
Received dns UDP packet of size 864, ifindex=2, ttl=0, fragsize=0, sender=2a01:cb00:59:f102::1, destination=2a01:cb00:59:f102:3905:14fb:c301:5421
Processing incoming packet of size 864 on transaction 63618 (rcode=SUCCESS).
Requesting DS to validate transaction 63618 (., DNSKEY with key tag: 9799).
Requesting DS to validate transaction 63618 (., DNSKEY with key tag: 20326).
Validating response from transaction 63618 (. IN DNSKEY).
Looking at . IN DNSKEY 256 3 RSASHA256 AwEAAZym4HCWiTAAl2Mv1izgTyn9sKwgi5eBxpG29bVlefq/r+TGCtmUElvFyBWHRjvf9mBglIlTBRse22dvzNOI+cYrkjD6LOHuxMoc/
                            d4WtXWKdviNmrtWF2GpjmDOI98gLd4BZ0U/lY847mJP9LypFABZcEn3zM3vce4Ee1A3upSlFQ2TFyJSD9HvMnP4XneFexBxV96RpLcy2O
                            +u2W6ChIiDCjlrowPCcU3zXfXxyWy/VKM6TOa8gNf+aKaVkcv/eIh5er8rrsqAi9KT8O5hmhzYLkUOQEXVSRORV0RMt9l3JSwWxT1MebE
                            DvtfBag3uo+mZwWSFlpc9kuzyWBd72Ec=
        -- Flags: ZONE_KEY
        -- Key tag: 9799: validated
Found verdict for lookup . IN DNSKEY: secure
Added positive authenticated non-confidential cache entry for . IN DNSKEY 7200s on enp9s0/INET6/2a01:cb00:59:f102::1
Added positive authenticated non-confidential cache entry for . IN DNSKEY 7200s on enp9s0/INET6/2a01:cb00:59:f102::1
Regular transaction 63618 for <. IN DNSKEY> on scope dns on enp9s0/* now complete with <success> from network (authenticated; non-confidential).
Validating response from transaction 6338 (io IN DS).
Looking at io IN DS 57355 8 2 95a57c3bab7849dbcddf7c72ada71a88146b141110318ca5be672057e865c3e2: validated
Found verdict for lookup io IN DS: secure
Added positive authenticated non-confidential cache entry for io IN DS 7200s on enp9s0/INET6/2a01:cb00:59:f102::1
Regular transaction 6338 for <io IN DS> on scope dns on enp9s0/* now complete with <success> from network (authenticated; non-confidential).
Validating response from transaction 28936 (io IN DNSKEY).
Looking at io IN DNSKEY 256 3 RSASHA256 AwEAAdt2Umpc+HdJPAhsPyeW9X08fDguO3zoSb9ajtZqhyc+VtbDwBCKo69QYpmop6ZctK9BgDtRMfd4TEqw4E/i1CrvDS5SFCuUqat6
                             /Np7zx+3UCkh2zhMrm3t0gS0VnoJqQ2q1ZZhnldzXOSX8JOYiIi7L3rvwnBjh33B+yLiu7fR
        -- Flags: ZONE_KEY
        -- Key tag: 28626: validated
Found verdict for lookup io IN DNSKEY: secure
Added positive authenticated non-confidential cache entry for io IN DNSKEY 3571s on enp9s0/INET6/2a01:cb00:59:f102::1
Added positive authenticated non-confidential cache entry for io IN DNSKEY 3571s on enp9s0/INET6/2a01:cb00:59:f102::1
Added positive authenticated non-confidential cache entry for io IN DNSKEY 3571s on enp9s0/INET6/2a01:cb00:59:f102::1
Regular transaction 28936 for <io IN DNSKEY> on scope dns on enp9s0/* now complete with <success> from network (authenticated; non-confidential).
Validating response from transaction 50553 (nextdns.io IN DS).
Looking at nextdns.io IN DS 2371 13 2 fa8e6c97f9769d8fbef6c694b28a2de882914784a2df171a376acd1f8168a633: validated
Found verdict for lookup nextdns.io IN DS: secure
Added positive authenticated non-confidential cache entry for nextdns.io IN DS 2260s on enp9s0/INET6/2a01:cb00:59:f102::1
Regular transaction 50553 for <nextdns.io IN DS> on scope dns on enp9s0/* now complete with <success> from network (authenticated; non-confidential).
Validating response from transaction 22752 (nextdns.io IN DNSKEY).
Looking at nextdns.io IN DNSKEY 256 3 ECDSAP256SHA256 oJMRESz5E4gYzS/q6XDrvU1qMPYIjCWzJaOau8XNEZeqCYKD5ar0IRd8KqXXFJkqmVfRvMGPmM1x8fGAa2XhSA==
        -- Flags: ZONE_KEY
        -- Key tag: 34505: validated
Found verdict for lookup nextdns.io IN DNSKEY: secure
Added positive authenticated non-confidential cache entry for nextdns.io IN DNSKEY 2260s on enp9s0/INET6/2a01:cb00:59:f102::1
Added positive authenticated non-confidential cache entry for nextdns.io IN DNSKEY 2260s on enp9s0/INET6/2a01:cb00:59:f102::1
Regular transaction 22752 for <nextdns.io IN DNSKEY> on scope dns on enp9s0/* now complete with <success> from network (authenticated; non-confidential).
Validating response from transaction 22015 (dns.nextdns.io IN AAAA).
Looking at dns.nextdns.io IN CNAME steering.nextdns.io: validated
Found verdict for lookup dns.nextdns.io IN CNAME: secure
Looking at steering.nextdns.io IN AAAA 2a0e:9900:0:1::1:2: no-signature
Added positive authenticated non-confidential cache entry for dns.nextdns.io IN CNAME 37s on enp9s0/INET6/2a01:cb00:59:f102::1
Regular transaction 22015 for <dns.nextdns.io IN AAAA> on scope dns on enp9s0/* now complete with <success> from network (authenticated; non-confidential).
Validating response from transaction 19450 (dns.nextdns.io IN A).
Looking at dns.nextdns.io IN CNAME steering.nextdns.io: validated
Found verdict for lookup dns.nextdns.io IN CNAME: secure
Looking at steering.nextdns.io IN A 193.168.204.73: no-signature
Added positive authenticated non-confidential cache entry for dns.nextdns.io IN CNAME 37s on enp9s0/INET6/2a01:cb00:59:f102::1
Regular transaction 19450 for <dns.nextdns.io IN A> on scope dns on enp9s0/* now complete with <success> from network (authenticated; non-confidential).
Freeing transaction 22015.
Following CNAME/DNAME dns.nextdns.io → steering.nextdns.io.
Cache miss for steering.nextdns.io IN A
Firing regular transaction 53398 for <steering.nextdns.io IN A> scope dns on enp9s0/* (validate=yes).
Using feature level UDP+EDNS0+DO for transaction 53398.
Using DNS server 2a01:cb00:59:f102::1 for transaction 53398.
Announcing packet size 1452 in egress EDNS(0) packet.
Emitting UDP, link MTU is 1500, socket MTU is 0, minimal MTU is 60
Sending query packet with id 53398 of size 71.
Cache miss for steering.nextdns.io IN AAAA
Firing regular transaction 47901 for <steering.nextdns.io IN AAAA> scope dns on enp9s0/* (validate=yes).
Using feature level UDP+EDNS0+DO for transaction 47901.
Using DNS server 2a01:cb00:59:f102::1 for transaction 47901.
Announcing packet size 1452 in egress EDNS(0) packet.
Emitting UDP, link MTU is 1500, socket MTU is 0, minimal MTU is 60
Sending query packet with id 47901 of size 71.
Freeing transaction 19450.
Freeing transaction 22752.
Freeing transaction 50553.
Freeing transaction 28936.
Freeing transaction 6338.
Freeing transaction 63618.
Received dns UDP packet of size 80, ifindex=2, ttl=0, fragsize=0, sender=2a01:cb00:59:f102::1, destination=2a01:cb00:59:f102:3905:14fb:c301:5421
Processing incoming packet of size 80 on transaction 53398 (rcode=SUCCESS).
Requesting SOA to validate transaction 53398 (steering.nextdns.io, unsigned non-SOA/NS RRset <steering.nextdns.io IN A 37.252.225.79>).
Cache miss for steering.nextdns.io IN SOA
Firing regular transaction 9851 for <steering.nextdns.io IN SOA> scope dns on enp9s0/* (validate=yes).
Using feature level UDP+EDNS0+DO for transaction 9851.
Using DNS server 2a01:cb00:59:f102::1 for transaction 9851.
Announcing packet size 1452 in egress EDNS(0) packet.
Emitting UDP, link MTU is 1500, socket MTU is 0, minimal MTU is 60
Sending query packet with id 9851 of size 71.
Requesting SOA to validate transaction 53398 (steering.nextdns.io, unsigned non-SOA/NS RRset <steering.nextdns.io IN A 193.168.204.73>).
Received dns UDP packet of size 104, ifindex=2, ttl=0, fragsize=0, sender=2a01:cb00:59:f102::1, destination=2a01:cb00:59:f102:3905:14fb:c301:5421
Processing incoming packet of size 104 on transaction 47901 (rcode=SUCCESS).
Requesting SOA to validate transaction 47901 (steering.nextdns.io, unsigned non-SOA/NS RRset <steering.nextdns.io IN AAAA 2a00:11c0:2:998::3>).
Requesting SOA to validate transaction 47901 (steering.nextdns.io, unsigned non-SOA/NS RRset <steering.nextdns.io IN AAAA 2a0e:9900:0:1::1:2>).
Received dns UDP packet of size 48, ifindex=2, ttl=0, fragsize=0, sender=2a01:cb00:59:f102::1, destination=2a01:cb00:59:f102:3905:14fb:c301:5421
Processing incoming packet of size 48 on transaction 9851 (rcode=SUCCESS).
Requesting DS (→ steering.nextdns.io) to validate transaction 9851 (steering.nextdns.io, unsigned empty SOA/NS response).
Cache miss for steering.nextdns.io IN DS
Firing regular transaction 55150 for <steering.nextdns.io IN DS> scope dns on enp9s0/* (validate=yes).
Using feature level UDP+EDNS0+DO for transaction 55150.
Using DNS server 2a01:cb00:59:f102::1 for transaction 55150.
Announcing packet size 1452 in egress EDNS(0) packet.
Emitting UDP, link MTU is 1500, socket MTU is 0, minimal MTU is 60
Sending query packet with id 55150 of size 71.
Received dns UDP packet of size 364, ifindex=2, ttl=0, fragsize=0, sender=2a01:cb00:59:f102::1, destination=2a01:cb00:59:f102:3905:14fb:c301:5421
Processing incoming packet of size 364 on transaction 55150 (rcode=SUCCESS).
Requesting DNSKEY to validate transaction 55150 (steering.nextdns.io, RRSIG with key tag: 34505).
Positive cache hit for nextdns.io IN DNSKEY
Regular transaction 50786 for <nextdns.io IN DNSKEY> on scope dns on enp9s0/* now complete with <success> from cache (authenticated; non-confidential).
Requesting DNSKEY to validate transaction 55150 (nextdns.io, RRSIG with key tag: 34505).
Validating response from transaction 55150 (steering.nextdns.io IN DS).
Looking at steering.nextdns.io IN NSEC steering\000.nextdns.io ( NS RRSIG NSEC ): validated
Found verdict for lookup steering.nextdns.io IN NSEC: secure
Looking at nextdns.io IN SOA dawn.ns.cloudflare.com dns.cloudflare.com 2267665164 10000 2400 604800 3600: validated
Found verdict for lookup nextdns.io IN SOA: secure
Proved NODATA via NSEC/NSEC3 for transaction 55150 (steering.nextdns.io IN DS)
Found verdict for lookup steering.nextdns.io IN DS: secure
Added positive authenticated non-confidential cache entry for nextdns.io IN SOA 2260s on enp9s0/INET6/2a01:cb00:59:f102::1
Added NODATA cache entry for steering.nextdns.io IN DS 2260s
Regular transaction 55150 for <steering.nextdns.io IN DS> on scope dns on enp9s0/* now complete with <success> from network (authenticated; non-confidential).
Validating response from transaction 9851 (steering.nextdns.io IN SOA).
Found verdict for lookup steering.nextdns.io IN SOA: bogus
DNSSEC validation failed for question steering.nextdns.io IN SOA: no-signature
Regular transaction 9851 for <steering.nextdns.io IN SOA> on scope dns on enp9s0/* now complete with <dnssec-failed> from network (unsigned; non-confidential).
Auxiliary DNSSEC RR query failed validation: no-signature
DNSSEC validation failed for question steering.nextdns.io IN A: no-signature
Regular transaction 53398 for <steering.nextdns.io IN A> on scope dns on enp9s0/* now complete with <dnssec-failed> from network (unsigned; non-confidential).
Auxiliary DNSSEC RR query failed validation: no-signature
DNSSEC validation failed for question steering.nextdns.io IN AAAA: no-signature
Regular transaction 47901 for <steering.nextdns.io IN AAAA> on scope dns on enp9s0/* now complete with <dnssec-failed> from network (unsigned; non-confidential).
Freeing transaction 53398.
Sent message type=error sender=n/a destination=:1.6292 path=n/a interface=n/a member=n/a cookie=33 reply_cookie=2 signature=s error-name=org.freedesktop.resolve1.DnssecFailed error-message=DNSSEC validation failed: no-signature
Sent message type=method_call sender=n/a destination=org.freedesktop.DBus path=/org/freedesktop/DBus interface=org.freedesktop.DBus member=RemoveMatch cookie=34 reply_cookie=0 signature=s error-name=n/a error-message=n/a
Freeing transaction 47901.
Freeing transaction 9851.
Freeing transaction 55150.
Freeing transaction 50786.
```

To compare, here is the output of `delv`:

```text
 17:52 ❱ delv +rtrace +vtrace A dns.nextdns.io @8.8.8.8
;; fetch: dns.nextdns.io/A
;; validating dns.nextdns.io/CNAME: starting
;; validating dns.nextdns.io/CNAME: attempting positive response validation
;; fetch: nextdns.io/DNSKEY
;; validating nextdns.io/DNSKEY: starting
;; validating nextdns.io/DNSKEY: attempting positive response validation
;; fetch: nextdns.io/DS
;; validating nextdns.io/DS: starting
;; validating nextdns.io/DS: attempting positive response validation
;; fetch: io/DNSKEY
;; validating io/DNSKEY: starting
;; validating io/DNSKEY: attempting positive response validation
;; fetch: io/DS
;; validating io/DS: starting
;; validating io/DS: attempting positive response validation
;; fetch: ./DNSKEY
;; validating ./DNSKEY: starting
;; validating ./DNSKEY: attempting positive response validation
;; validating ./DNSKEY: verify rdataset (keyid=20326): success
;; validating ./DNSKEY: marking as secure (DS)
;; validating io/DS: in fetch_callback_dnskey
;; validating io/DS: keyset with trust secure
;; validating io/DS: resuming validate
;; validating io/DS: verify rdataset (keyid=9799): success
;; validating io/DS: marking as secure, noqname proof not needed
;; validating io/DNSKEY: in fetch_callback_ds
;; validating io/DNSKEY: dsset with trust secure
;; validating io/DNSKEY: verify rdataset (keyid=57355): success
;; validating io/DNSKEY: marking as secure (DS)
;; validating nextdns.io/DS: in fetch_callback_dnskey
;; validating nextdns.io/DS: keyset with trust secure
;; validating nextdns.io/DS: resuming validate
;; validating nextdns.io/DS: verify rdataset (keyid=28626): success
;; validating nextdns.io/DS: marking as secure, noqname proof not needed
;; validating nextdns.io/DNSKEY: in fetch_callback_ds
;; validating nextdns.io/DNSKEY: dsset with trust secure
;; validating nextdns.io/DNSKEY: verify rdataset (keyid=2371): success
;; validating nextdns.io/DNSKEY: marking as secure (DS)
;; validating dns.nextdns.io/CNAME: in fetch_callback_dnskey
;; validating dns.nextdns.io/CNAME: keyset with trust secure
;; validating dns.nextdns.io/CNAME: resuming validate
;; validating dns.nextdns.io/CNAME: verify rdataset (keyid=34505): success
;; validating dns.nextdns.io/CNAME: marking as secure, noqname proof not needed
;; fetch: steering.nextdns.io/A
;; validating steering.nextdns.io/A: starting
;; validating steering.nextdns.io/A: attempting insecurity proof
;; validating steering.nextdns.io/A: checking existence of DS at 'io'
;; validating steering.nextdns.io/A: checking existence of DS at 'nextdns.io'
;; validating steering.nextdns.io/A: checking existence of DS at 'steering.nextdns.io'
;; fetch: steering.nextdns.io/DS
;; validating steering.nextdns.io/DS: starting
;; validating steering.nextdns.io/DS: attempting negative response validation from message
;;   validating nextdns.io/SOA: starting
;;   validating nextdns.io/SOA: attempting positive response validation
;;   validating nextdns.io/SOA: keyset with trust secure
;;   validating nextdns.io/SOA: verify rdataset (keyid=34505): success
;;   validating nextdns.io/SOA: marking as secure, noqname proof not needed
;; validating steering.nextdns.io/DS: in validator_callback_nsec
;; validating steering.nextdns.io/DS: resuming validate_nx
;;   validating steering.nextdns.io/NSEC: starting
;;   validating steering.nextdns.io/NSEC: attempting positive response validation
;;   validating steering.nextdns.io/NSEC: keyset with trust secure
;;   validating steering.nextdns.io/NSEC: verify rdataset (keyid=34505): success
;;   validating steering.nextdns.io/NSEC: marking as secure, noqname proof not needed
;; validating steering.nextdns.io/DS: in validator_callback_nsec
;; validating steering.nextdns.io/DS: looking for relevant NSEC
;; validating steering.nextdns.io/DS: nsec proves name exists (owner) data=0
;; validating steering.nextdns.io/DS: resuming validate_nx
;; validating steering.nextdns.io/DS: nonexistence proof(s) found
;; validating steering.nextdns.io/A: in fetch_callback_ds
;; validating steering.nextdns.io/A: marking as answer (fetch_callback_ds)
; fully validated
dns.nextdns.io.         300     IN      CNAME   steering.nextdns.io.
dns.nextdns.io.         300     IN      RRSIG   CNAME 13 3 300 20220116175253 20220114155253 34505 nextdns.io. Wxyi2P8hymIcSh2Mg/rGQQrmXt5c7jEGMhsP8tGOw0o0LUxejexOSFJR nMfG369nQHG36U4YFBP44Ok4SwbSTA==

; unsigned answer
steering.nextdns.io.    60      IN      A       37.252.225.79
steering.nextdns.io.    60      IN      A       193.168.204.73
```

**Comment by vincentbernat** [comment]
FI, a SOA record is being added to check if it works around the issue. If you try to reproduce it, use `abcd.test.nextdns.io` which won't get a SOA and has the same problem.

**Comment by vincentbernat** [comment]
After adding a SOA record for steering.nextdns.io, resolved is able to do the verification. `abcd.test.nextdns.io` is still broken while valid.

**Comment by frittentheke** [comment]
I just ran into this problem when debugging dnssec failures (for non DNSSEC enabled zones). 

In my case they are reported on reverse zones (PTR) for dynamically assigned address space of an ISP. For those there one receives `REFUSED` for SOA or NS queries and only queries for a particular PTR actually return results.

Since the subdomain tree for those IPv6 addresses is quite deep, there are lots of queries issued and and in the end dnssec comes back as failed (and not insecure) there is no result.

**Comment by rpigott** [comment]
Resolved by #31827.
